src/Service/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -194,5 +197,59 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
 
             return sessionMapQuery.ToString();
         }
+
+        /// <inheritdoc/>
+        public override async Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
+            DbDataReader dbDataReader, List<string>? args = null)
+        {
+            // From the first result set, we get the count(0/1) of records with given PK.
+            DbResultSet resultSetWithCountOfRowsWithGivenPk = await ExtractResultSetFromDbDataReader(dbDataReader);
+            int numOfRecordsWithGivenPK = (int)resultSetWithCountOfRowsWithGivenPk.Rows.FirstOrDefault()!.Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]!;
+
+            // The second result set holds the records returned as a result of the executed update/insert operation.
+            DbResultSet? dbResultSet = await dbDataReader.NextResultAsync() ? await ExtractResultSetFromDbDataReader(dbDataReader) : null;
+
+            if (dbResultSet is null)
+            {
+                // This can only happen in the case where we are dealing with table/view with auto-gen PK.
+                // In this case, insert cannot happen.
+                // Also since dbResultSet is null, update couldn't execute as well because there was no record corresponding to given PK.
+                string prettyPrintPk = args![0];
+                string entityName = args[1];
+
+                throw new DataApiBuilderException(
+                        message: $"Cannot perform INSERT and could not find {entityName} " +
+                        $"with primary key {prettyPrintPk} to perform UPDATE on.",
+                        statusCode: HttpStatusCode.NotFound,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.EntityNotFound);
+
+            }
+
+            if (numOfRecordsWithGivenPK == 1) // This indicates that a record existed with given PK and we attempted an update operation.
+            {
+                if (dbResultSet.Rows.Count == 0)
+                {
+                    // Record exists in the table/view but no record updated - indicates database policy failure.
```

Comment:
>// Record exists in the table/view but no record updated - indicates database policy failure. [](http://example.com/codeflow?start=20&length=93)

nit: for future readers:  indicates database policy failure because xyz

Reply:
When a record exists but is not updated, it indicates a db policy failure. What else should we add here?

Reply:
should be fine after re-reading. Originally I was thinking something additional like "because a db policy restricted the record from being in scope of the update operation"